### PR TITLE
Update Bootstrap: Various fixes and specs

### PR DESCRIPTION
**Pull request by maiwald**

### CODE REVIEW

File: app/packs/src/apps/mydb/elements/details/reactions/schemeTab/Material.js [hunk 1]
```diff
@@ -291,7 +291,7 @@ class Material extends Component {
       <td colSpan={colSpan} style={style}>
         <NumeralInputWithUnitsCompo
           precision={4}
-          bsStyle="success"
+          variant="success"
```
Comment (jhmegorei) [inline app/packs/src/apps/mydb/elements/details/reactions/schemeTab/Material.js:294]: bsStyle wasn't used in NumeralInputWithUnitsCompo before, so changing it to variant here would change the behaviour imo.

Comment (jhmegorei) [reply app/packs/src/apps/mydb/elements/details/reactions/schemeTab/Material.js:294]: Ok after looking at the file history, it looks like Joachim missed some occurrences when replacing bsStyle, so it should be allright

File: app/packs/src/apps/admin/ThirdPartyApp.js [hunk 2]
```diff
@@ -266,7 +266,7 @@ export default class ThirdPartyApp extends React.Component {
         <Modal.Body style={{ overflow: 'auto' }}>
           <div className="col-md-12">
             <FormGroup controlId="formControlName">
-              <Col componentClass={ControlLabel} sm={3}>
+              <Col as={ControlLabel} sm={3}>
```
Comment (jhmegorei) [inline app/packs/src/apps/admin/ThirdPartyApp.js:269]: ControlLabel is a legacy Component and should be replaced with Form.Label

File: app/packs/src/apps/mydb/elements/details/researchPlans/ImageAnnotationEditButton.js [hunk 3]
```diff
@@ -7,7 +7,19 @@ import Attachment from 'src/models/Attachment';
 export default class ImageAnnotationEditButton extends Component {
   allowedFileTypes = ['png', 'jpg', 'bmp', 'tif', 'svg', 'jpeg', 'tiff'];
 
-  renderButton(isActive, tooltipText) {
+  render() {
```
Comment (jhmegorei) [inline app/packs/src/apps/mydb/elements/details/researchPlans/ImageAnnotationEditButton.js:10]: Apart from personal preference, what did you change here? I see no bootstrap related changes.

File: app/packs/src/components/UserLabels.js [hunk 4]
```diff
@@ -227,7 +227,7 @@ class UserLabelModal extends Component {
     return (
       <Form horizontal>
         <FormGroup controlId="accessLevelInput">
-          <Col componentClass={ControlLabel} sm={2}>
+          <Col as={ControlLabel} sm={2}>
```
Comment (jhmegorei) [inline app/packs/src/components/UserLabels.js:230]: See above, ControlLabel -> Form.Label

File: app/packs/src/components/common/ChevronIcon.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
-import React from 'react';
+import React, { forwardRef } from 'react';
```
Comment (jhmegorei) [inline app/packs/src/components/common/ChevronIcon.js:1]: What is forwardRef and why is it needed here?

File: app/packs/src/apps/admin/ThirdPartyApp.js (hunk 2 above)
Comment (maiwald) [reply app/packs/src/apps/admin/ThirdPartyApp.js:269]: I only wanted to get rid of the warning regarding the unknown `componentClass` prop. @ashwiniHerle has marked this and the `UserLabels` component as pending so I did not want to change too much.

**Pull request merged**